### ISSUE TITLE
Enhance `readableColor` to support preferences

### ISSUE THREAD

**Issue by mrmckeb** [styled-components/polished#479]
We've been using Polished more and more in our current project, and find it particularly useful for accessibility.

**The problem**

`readableColor` will return a color that is not accessible whenever the user-supplied `lightReturnColor`/`darkReturnColor` does not contrast strongly enough with `color`.

This is not an issue if the user does not supply `lightReturnColor`/`darkReturnColor` as black/white are returned.

**Proposal**

To resolve this, I propose that we add `strict` flag as a fourth parameter, that ensures the function always returns an accessible color (as per WCAG AA).

```ts
readableColor(color: string, lightReturnColor: string, darkReturnColor: string, strict: boolean)
```

With this flag, if either`lightReturnColor` or `darkReturnColor` do not meet contrast requirements against a given color, `black` or `white` would be returned instead.

This would not be a breaking change.

**Possible change for next major**

In the next major release of polished, this _could_ become the default setting and `strict` could be removed.

---

If you think this change is acceptable, I'll raise a PR. Feedback and other thoughts welcome.

**Comment by bhough** [comment]
@mrmckeb Definitely like this idea and agree with the staged rollout approach. We would gladly accept a PR for this.

**Comment by mrmckeb** [comment]
Thanks @bhough, I've created PR #484 and welcome any feedback.

**Comment by mrmckeb** [comment]
@bhough should I close this off now? Or will you close when you release? Thanks again.

**Comment by bhough** [comment]
@mrmckeb We will close it once it is actually released.
